### ISSUE TITLE
Document clone alters original document

### ISSUE THREAD

**Issue by TyLindberg** [donmccurdy/glTF-Transform#603]
**Describe the bug**

Calling `Document.clone` on the ToyCar example model results in KHR_texture_transform metadata being removed from the materials on the original document. The extension will be preserved in the cloned document.

This seems to be an issue with the `merge` method based on the [implementation for `clone`](https://github.com/donmccurdy/glTF-Transform/blob/bc4a48e025187b2b01dc32509573ab6ded3b8fe9/packages/core/src/document.ts#L120). Hence this may affect other operations outside of `clone`.

**To Reproduce**

I used the following code as a test case.

```js
import { NodeIO } from "@gltf-transform/core";
import { ALL_EXTENSIONS } from "@gltf-transform/extensions";

(async () => {
  const io = new NodeIO().registerExtensions(ALL_EXTENSIONS);
  const doc = await io.read("ToyCar.glb");

  const clone = doc.clone();

  await io.write("ToyCarOriginalClone.glb", doc);
  await io.write("ToyCarClone.glb", clone);
})();
```

**Expected behavior**

I would expect the original document to be unmodified when calling `clone` and `merge`.

**Versions:**
 - Version: v2.1.6
 - Environment: Node.js 14

**Comment by TyLindberg** [comment]
Great! Thanks for taking a look!

**Comment by donmccurdy** [comment]
Fix published to v﻿﻿2.1.7 — sorry this bug must've been a headache to find!

**Comment by TyLindberg** [comment]
No worries! Thanks for the quick fix!
